main.cpp

```diff
@@ -42,16 +42,59 @@ struct Point
     }
 private:
     float x{0}, y{0};
-};
+};
 
 template<typename Type>
 struct Wrapper
 {
     Wrapper(Type&& t) : val(std::move(t)) 
     { 
         std::cout << "Wrapper(" << typeid(val).name() << ")" << std::endl; 
-    }
-};
+    }
+
+    void print()
+    {
+        std::cout << "Wrapper::print(" << val << ")" << std::endl;
+    }
+    
+private:
+    Type val{0};
+
+};
+
+    //I'm expecting to see a separate class definition entirely, like you did with the Numeric<double> in project 4.
+template<>
+struct Wrapper<Point>
+{
+    Wrapper(Point&& t) : val(std::move(t)) 
+    { 
+        std::cout << "Wrapper(" << typeid(val).name() << ")" << std::endl; 
+    }
+
+    void print()
+    {
+        std::cout << "Wrapper::print(" << val.toString() << ")" << std::endl;
+    }
+    
+private:
+    Point val{0.0f, 0.0f};
+};
```

Comment:
Project 9.1 Challenge 1:
There is a lot of repetition between the template class `Wrapper`, and the template specialization for `Wrapper<Point>`
Can you figure out how to eliminate the duplicate constructor?
hint: you only have to _explicitly specialize_ that which is unique.
You can specialize member functions without specializing the entire class :wink: